### PR TITLE
[automate-2216] Supply missing field caption for team details

### PR DESCRIPTION
**Pull request by msorens**
### :nut_and_bolt: Description: What code changed, and why?

A bug was recently introduced that caused a field caption for team details
to disappear:
 
image

### :+1: Definition of Done

Caption now appears on V2:
image

Caption now appears on V1:
image

### :athletic_shoe: How to Build and Test the Change

rebuild automate-ui
Navigate to Settings >> Teams >> [team] >> Details
Try on both v1 and v2

### CODE REVIEW

File: components/automate-ui/src/app/pages/team-details/team-details.component.ts [hunk 1]
```diff
@@ -100,6 +100,10 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
     this.store.dispatch(new GetUsers());
 
     this.isIAMv2$ = this.store.select(isIAMv2);
+
+    this.isIAMv2$.pipe(takeUntil(this.isDestroyed))
+      .subscribe((isV2) => this.descriptionOrName = isV2 ? 'name' : 'description');
```
Comment (msorens) [inline components/automate-ui/src/app/pages/team-details/team-details.component.ts]: This got lost in the shuffle during recent refactoring--restoring it.

File: components/automate-ui/src/app/pages/team-details/team-details.component.ts [hunk 2]
```diff
@@ -132,13 +136,7 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
...
     ]).pipe(
       filter(([isV2, _v1TeamFromRoute, _v2TeamFromRoute]) => isV2 !== null),
-      map(([isV2, v1Team, v2Team]) => {
-        if (isV2) {
-          return v2Team;
-        } else {
-          return v1Team;
-        }
-      }));
+      map(([isV2, v1Team, v2Team]) => isV2 ? v2Team : v1Team ));
```
Comment (msorens) [inline components/automate-ui/src/app/pages/team-details/team-details.component.ts]: Trivial optimization--unrelated.

**Comment by blakestier** [comment]
@msorens does this also take care of the error message?

**Comment by msorens** [comment]
@blakestier : Yes, should have said. I have updated the images to show it.

**Comment by susanev** [comment]
yup 30px of space, @msorens is gonna add it when hes back

**Pull request merged**